inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -591,19 +591,21 @@ void RemovePermutationsNHWCToNCHWPass::run() {
     for (auto&& toRemove : permutationsToRemove) {
         gnalog() << toRemove->type << " layer '" << toRemove->name << "' will be removed" << '\n';
 
-        auto next = getInputTo(toRemove->outData.front()).begin()->second;
-        if (LayerInfo(next).isConvolution()) {
-            next->input()->setDims(toRemove->input()->getDims());
-            next->input()->setLayout(Layout::NHWC);
-            auto layerBeforePermute = CNNNetPrevLayer(toRemove);
-            layerBeforePermute->outData[0]->setLayout(Layout::NHWC);
-
-            auto& convolution = dynamic_cast<ConvolutionLayer&>(*next);
-            if (convolution._kernel_y != 1) {
-                THROW_GNA_LAYER_EXCEPTION(next) << "this case is not implemented yet";
+        if (getInputTo(toRemove->outData.front()).size()) {
+            auto next = getInputTo(toRemove->outData.front()).begin()->second;
+            if (LayerInfo(next).isConvolution()) {
+                next->input()->setDims(toRemove->input()->getDims());
+                next->input()->setLayout(Layout::NHWC);
+                auto layerBeforePermute = CNNNetPrevLayer(toRemove);
+                layerBeforePermute->outData[0]->setLayout(Layout::NHWC);
+
+                auto& convolution = dynamic_cast<ConvolutionLayer&>(*next);
```

Comment:
Need to add assert after this - otherwise static checking will complain.

Reply:
I didn't notice this is a reference, not a pointer, so my comment is invalid. In that case, I don't understand what's the point to use dynamic_cast instead of static_cast?

Reply:
the only benefit of static_cast in that case is speed which not a bottleneck here.

Reply:
But what is the reason to use dynamic_cast at all if the result is not compared with anything? If we are sure the cast will be successful, let's use dynamic_cast. If we are not quite sure, let's use a pointer and add a check or assert.

Reply:
We discussed this with @esmirno, and are proposing to use dynamic cast, pointer, and add a check which would produce a reasonable message if the conversion is not successful.